### ISSUE TITLE
[Feature] make query validation collect all errors instead of throwing on each one

### ISSUE THREAD

**Issue by tlento** [dbt-labs/metricflow#776]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing metricflow functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Our query validator iterates through a bunch of properties and parameters and throws an exception on every error it sees, which means it throws an exception on the first error it sees. So a user with, say, 3 mistakes gets to do this:

```
Run query
Hit error
Update query
Run query
Hit error
Update query
Run query
Hit error
Update query
Run query
FINALLY!
```

It sure would be nicer if this would look more like:

```
Run query
Hit error with complete listing of issues
Update query
Success!
```

See [discussion on tangentially related PR](https://github.com/dbt-labs/metricflow/pull/772/files#r1327507085) for more context.

**Comment by Jstein77** [comment]
Yes please!

**Comment by tlento** [comment]
Paul fixed this in #914
